charts/kong/templates/_helpers.tpl

```diff
@@ -492,12 +492,15 @@ The name of the service used for the ingress controller's validation webhook
 {{- end -}}
 
 {{- define "kong.wait-for-db" -}}
+{{ $sockFile := (printf "%s/stream_rpc.sock" (default "/usr/local/kong" .Values.env.prefix)) }}
 - name: wait-for-db
   image: {{ include "kong.getRepoTag" .Values.image }}
   imagePullPolicy: {{ .Values.image.pullPolicy }}
   env:
   {{- include "kong.env" . | nindent 2 }}
-  command: [ "/bin/sh", "-c", "until kong start; do echo 'waiting for db'; sleep 1; done; kong stop" ]
+{{/* TODO: the rm command here is a workaround for https://github.com/Kong/charts/issues/295
+     It should be removed once that's fixed */}}
+  command: [ "/bin/sh", "-c", "until kong start; do echo 'waiting for db'; sleep 1; done; kong stop; if stat {{ $sockFile }} > /dev/null; then rm {{ $sockFile }}; else true; fi"]
```

Comment:
```suggestion
  command: [ "/bin/sh", "-c", "until kong start; do echo 'waiting for db'; sleep 1; done; kong stop; rm -fv {{ $sockFile | squote }}"]
```

`-f` makes `rm` _ignore nonexistent files and arguments_ and _never prompt_. Both seem appropriate in this case. Also without the `if` the code is less verbose.
Also quoted the sockFile. Without quotes, if the filename contains a space character, it will be expanded into multiple arguments for `rm`.